Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -312,4 +318,56 @@ extension BrowserViewController: ReaderModeDelegate {
             toolbar.updateReaderModeState(state)
         }
     }
+}
+
+extension BrowserViewController: BrowserLongPressGestureDelegate {
+    func browser(browser: Browser, elementsDidLongPressed elements: [ElementType]) {
+        var actionSheetController = UIAlertController(title: "", message: "Mozilla Firefox", preferredStyle: UIAlertControllerStyle.ActionSheet)
+        var url: NSURL?
+        for element in elements {
+            switch element {
+            case .hrefLink(let urlLink):
+                url = urlLink
+                var openNewTabAction =  UIAlertAction(title: "Open In New Tab", style: UIAlertActionStyle.Default) { (action: UIAlertAction!) in
+                    let request =  NSURLRequest(URL:urlLink)
+                    let tab = self.tabManager.addTab(request: request)
+                    return ()  //need explicit void return
+                }
+                actionSheetController.addAction(openNewTabAction)
+
+                var copyAction = UIAlertAction(title: "Copy", style: UIAlertActionStyle.Default) { (action:UIAlertAction!) -> Void in
+                    var pasteBoard = UIPasteboard.generalPasteboard()
+                    pasteBoard.string = urlLink.absoluteString
+                }
+                actionSheetController.addAction(copyAction)
+
+            case .ImageSrc(let urlLink):
+                if url == nil {
+                    url = urlLink
+                }
+                var saveImageAction = UIAlertAction(title: "Save Image", style: UIAlertActionStyle.Default) { (action:UIAlertAction!) -> Void in
+                    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), { () -> Void in
+                        var imageData = NSData(contentsOfURL: urlLink)
+                        if imageData != nil {
+                            UIImageWriteToSavedPhotosAlbum(UIImage(data: imageData!), nil, nil, nil)
+                        }
+                    })
+                }
+                actionSheetController.addAction(saveImageAction)
+
+                var copyAction = UIAlertAction(title: "Copy Image URL", style: UIAlertActionStyle.Default) { (action:UIAlertAction!) -> Void in
+                    var pasteBoard = UIPasteboard.generalPasteboard()
+                    pasteBoard.string = urlLink.absoluteString
+                }
+                actionSheetController.addAction(copyAction)
+
+            default:
+                return
+            }
+        }
+        actionSheetController.title = "\(url!.scheme!)://\(url!.host!)/"
```

Comment:
Please change this to <code>"\(dialogTitleURL!.absoluteString!)"</code> so we show the actual URL of the element being tapped.